packages/salesforcedx-vscode-apex/src/commands/forceApexExecute.ts

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {
+  ExecuteAnonymousResponse,
+  ExecuteService
+} from '@salesforce/apex-node';
+import {
+  getRootWorkspacePath,
+  hasRootWorkspace,
+  LibraryCommandletExecutor,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from '@salesforce/salesforcedx-utils-vscode/out/src';
+import {
+  Command,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import {
+  CancelResponse,
+  CommandletExecutor,
+  ContinueResponse,
+  ParametersGatherer
+} from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { OUTPUT_CHANNEL } from '../constants';
+import { workspaceContext } from '../context';
+import { nls } from '../messages';
+
+type TempFile = { fileName: string };
+
+function getRange(lineNumber: string, columnNumber: string): vscode.Range {
+  const ln = Number(lineNumber);
+  const col = Number(columnNumber);
+  const pos = new vscode.Position(ln > 0 ? ln - 1 : 0, col > 0 ? col - 1 : 0);
+  return new vscode.Range(pos, pos);
+}
```

Comment:
Needed this to process diagnostics, it's shared by the deploy/retrieve commands in the core package. We could also move this to utils...thoughts?

Reply:
If it's shared by the SDR commands too then I think we could move it over to utils

Reply:
Having trouble moving it over with the tests since it requires the vscode api ugh. not having luck with proxyquire or anything either.